src/main/kotlin/com/quipalup/katydid/logentry/primaryadapter/httprest/CreateLogEntryEndpoint.kt

```diff
@@ -1,29 +1,36 @@
 package com.quipalup.katydid.logentry.primaryadapter.httprest
 
+import arrow.core.Either
+import arrow.core.flatMap
+import arrow.core.right
+import com.quipalup.katydid.logentry.application.CreateLogEntryByFieldCommand
+import com.quipalup.katydid.logentry.application.CreateLogEntryCommandHandler
+import com.quipalup.katydid.logentry.domain.LogEntryError
 import org.springframework.http.HttpStatus
 import org.springframework.web.bind.annotation.PostMapping
-import org.springframework.web.bind.annotation.RequestBody
 import org.springframework.web.bind.annotation.ResponseStatus
 import org.springframework.web.bind.annotation.RestController
 
 @RestController
-class CreateLogEntryEndpoint {
+internal class CreateLogEntryEndpoint(private val createLogEntryCommandHandler: CreateLogEntryCommandHandler) {
     @PostMapping("/log-entries")
     @ResponseStatus(HttpStatus.CREATED)
-    fun execute(@RequestBody newLog: LogEntryRequestDocument): LogEntryResponseDocument {
+    fun execute(): LogEntryResponseDocument {
+        return buildCreateRequest()
+            .flatMap { createLogEntryCommandHandler.execute(it) }
+            .fold(errorHandler()) { it }
+    }
+
+    private fun errorHandler(): (LogEntryError) -> LogEntryResponseDocument = { throw RuntimeException() }
+
+    private fun buildCreateRequest(): Either<LogEntryError, CreateLogEntryByFieldCommand> = CreateLogEntryByFieldCommand(
```

Comment:
```suggestion
    private fun buildCommand(document: LogEntryRequestDocument): Either<CreateLogEntryError, CreateLogEntryCommand> = CreateLogEntryCommand(
```
This function takes the document and transforms it into a `CreateLogEntryCommand`.